theano/scan_module/tests/test_scan.py

```diff
@@ -3967,8 +3971,16 @@ def onestep(x, x_tm4):
         updates = OrderedDict([(sharedvar, results[0][-1:])])
 
         f = theano.function([seq], results[1], updates=updates)
+
+        # This fails if scan uses wrongly the borrow flag
         assert numpy.all(exp_out == f(inp))
 
+        # This fails if Scan's infer_shape() is unable to remove the Scan
+        # node from the graph.
+        f_infershape = theano.function([seq], results[1].shape)
+        scan_nodes_infershape = scan_nodes_from_fct(f_infershape)
+        assert(len(scan_nodes_infershape) == 0)
```

Comment:
This fail in FAST_COMPIILE. I think we should enable the local_shape_to_shape_i in FAST_COMPILE. This is fast to execute, the feature is already registered and this can remove much computation.

in tensor/opt.py, register local_shape_to_shape_i  like  `@register_canonicalize('fast_compile')